R/connect.R

```diff
@@ -14,16 +19,65 @@
 #' instance <- connect("laminlabs/cellxgene")
 #' instance
 #' }
-connect <- function(slug) {
-  user_settings <- .settings_load__load_or_create_user_settings()
+connect <- function(slug = NULL) {
+  instance_settings <-
+    if (is.null(slug)) {
+      # if the slug is null, see if we can load the default instance
+      instance_file <- .settings_store__current_instance_settings_file()
 
-  owner_name <- .connect_get_owner_name_from_identifier(slug)
+      if (file.exists(instance_file)) {
+        .settings_load__load_instance_settings(instance_file)
+      } else {
+        cli_abort(
+          paste0(
+            "Could not load default instance. Either:\n",
+            " - Provide a slug. For example: `connect(\"laminlabs/cellxgene\")`)\n",
+            " - Set a default instance by running `lamin load <slug>`."
+          )
+        )
+      }
+    } else {
+      # if the slug is not null, try to load the instance from the local settings
+      owner_name <- .connect_get_owner_name_from_identifier(slug)
 
-  instance_settings <- .connect_get_instance_settings(
-    owner = owner_name$owner,
-    name = owner_name$name,
-    access_token = user_settings$access_token
-  )
+      instance_file <- .settings_store__instance_settings_file(
+        name = owner_name$name,
+        owner = owner_name$owner
+      )
+
+      if (file.exists(instance_file)) {
+        .settings_load__load_instance_settings(instance_file)
+      } else {
+        # try to load the user settings from the api
+        user_file <- .settings_store__current_user_settings_file()
+
+        if (!file.exists(user_file)) {
+          cli_abort(paste0(
+            "No default user or instance is loaded! Either:\n",
+            " - Call `lamin login` to set a default user.\n",
+            " - Call `lamin load <slug>` to set a default instance.\n"
+          ))
+        } else {
+          user_settings <- .settings_load__load_user_settings(user_file)
+
+          .connect_get_instance_settings(
+            owner = owner_name$owner,
+            name = owner_name$name,
+            access_token = user_settings$access_token
+          )
+        }
+      }
```

Comment:
If the instance settings file does not exist, we _could_ try running `lamin load ${slug}` ourselves, as opposed to throwing an error message. @lazappi wdyt?

Reply:
For now I would probably leave the setup stuff to the command line, just for easiness. If Python lets you do this then maybe it would be nice to add later.